File: android/app/src/main/kotlin/com/tbtrapp/components/ContactActionSheet.kt
package com.tbtrapp.components

import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Call
import androidx.compose.material.icons.filled.Message
import androidx.compose.material.icons.filled.Videocam
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.tbtrapp.data.AppContact

// Bottom sheet shown when a contact row is tapped: Message / Voice Call /
// Video Call. This component is intentionally dumb — it just renders the
// three actions and calls back to whatever the screen passed in. The
// caller decides what "Message"/"Voice Call"/"Video Call" actually do
// (in-app chat + WebRTC call for matched app users vs. SMS/dialer/WhatsApp
// fallback for non-app contacts) — that branching lives in the screen
// that shows this sheet, not here.
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun ContactActionSheet(
    contact: AppContact,
    onDismiss: () -> Unit,
    onMessage: (AppContact) -> Unit,
    onVoiceCall: (AppContact) -> Unit,
    onVideoCall: (AppContact) -> Unit
) {
    ModalBottomSheet(onDismissRequest = onDismiss) {
        Column(modifier = Modifier.padding(horizontal = 8.dp, vertical = 4.dp)) {

            Text(
                text = contact.name,
                fontWeight = FontWeight.Bold,
                modifier = Modifier.padding(start = 16.dp, bottom = 2.dp)
            )
            Text(
                text = contact.phone,
                color = Color.Gray,
                modifier = Modifier.padding(start = 16.dp, bottom = 12.dp)
            )

            ActionItem(icon = Icons.Default.Message, label = "Message") {
                onMessage(contact)
                onDismiss()
            }
            // 🔑 FIX: Voice Call and Video Call rows never existed — only
            // "Message" was rendered, even though the sheet's whole point
            // (per its own header comment) was Message/Voice/Video.
            ActionItem(icon = Icons.Default.Call, label = "Voice Call") {
                onVoiceCall(contact)
                onDismiss()
            }
            ActionItem(icon = Icons.Default.Videocam, label = "Video Call") {
                onVideoCall(contact)
                onDismiss()
            }

            Spacer(modifier = Modifier.height(12.dp))
        }
    }
}

@Composable
private fun ActionItem(
    icon: ImageVector,
    label: String,
    onClick: () -> Unit
) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .clickable { onClick() }
            .padding(horizontal = 16.dp, vertical = 14.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        Icon(imageVector = icon, contentDescription = label, tint = Color(0xFF365DDB))
        Spacer(modifier = Modifier.width(20.dp))
        Text(text = label, fontWeight = FontWeight.Medium)
    }
}

File: lib/screens/components/contact_action_sheet.dart
import 'package:flutter/material.dart';
import '../../data/app_contact.dart';

/// Equivalent of ContactActionSheet.kt. Same "dumb sheet" philosophy —
/// this only renders the three actions; the caller (ContactScreen)
/// decides what each one actually does.
Future<void> showContactActionSheet({
  required BuildContext context,
  required AppContact contact,
  required void Function(AppContact) onMessage,
  required void Function(AppContact) onVoiceCall,
  required void Function(AppContact) onVideoCall,
}) {
  return showModalBottomSheet(
    context: context,
    builder: (context) {
      return SafeArea(
        child: Padding(
          padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 4),
          child: Column(
            mainAxisSize: MainAxisSize.min,
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              Padding(
                padding: const EdgeInsets.only(left: 16, bottom: 2, top: 8),
                child: Text(contact.name, style: const TextStyle(fontWeight: FontWeight.bold)),
              ),
              Padding(
                padding: const EdgeInsets.only(left: 16, bottom: 12),
                child: Text(contact.phone, style: const TextStyle(color: Colors.grey)),
              ),
              _ActionItem(
                icon: Icons.message,
                label: 'Message',
                onTap: () {
                  Navigator.pop(context);
                  onMessage(contact);
                },
              ),
              _ActionItem(
                icon: Icons.call,
                label: 'Voice Call',
                onTap: () {
                  Navigator.pop(context);
                  onVoiceCall(contact);
                },
              ),
              _ActionItem(
                icon: Icons.videocam,
                label: 'Video Call',
                onTap: () {
                  Navigator.pop(context);
                  onVideoCall(contact);
                },
              ),
              const SizedBox(height: 12),
            ],
          ),
        ),
      );
    },
  );
}

class _ActionItem extends StatelessWidget {
  final IconData icon;
  final String label;
  final VoidCallback onTap;

  const _ActionItem({required this.icon, required this.label, required this.onTap});

  @override
  Widget build(BuildContext context) {
    return InkWell(
      onTap: onTap,
      child: Padding(
        padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 14),
        child: Row(
          children: [
            Icon(icon, color: const Color(0xFF365DDB)),
            const SizedBox(width: 20),
            Text(label, style: const TextStyle(fontWeight: FontWeight.w500)),
          ],
        ),
      ),
    );
  }
}